src/components/TextInput/TextInput.stories.mdx

```diff
@@ -104,6 +117,29 @@ is left blank, a value will be generated for you.
 
 <ArgsTable story="TextInput with Controls" />
 
+## Accessibility
+
+The `TextInput` component renders an HTML `<input>` element which is accessible
+via keyboard and screen reader. The `aria-required` and `required` attributes are
+set with the `isRequired` prop. The `aria-disabled` and `disabled` attributes are
+set with the `isDisabled` prop. The `aria-invalid` is set with the `isInvalid`
+prop.
+
+Internally, a `Label` is associated with the `<input>` element. When `showLabel`
```

Comment:
I like that you also mention what heppens with the `showLabel` prop.  This type of info should also be added to the Accessibility section for `CheckboxGroup`, `DatePicker`, `RadioGroup`, `Select` and `Slider`.